webpack/assets/javascripts/react_app/components/passwordStrength/index.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import CommonForm from '../common/forms/CommonForm';
+import ReactPasswordStrength from 'react-password-strength';
+import './password_strength.scss';
+
+class PasswordStrength extends React.Component {
+  constructor(props) {
+    super(props);
+    this.checkRetype = this.checkRetype.bind(this);
+    this.state = {
+      password: '',
+      error: props.data.verify.error
+    };
+  }
+  checkRetype(event) {
+    if (event.target.value !== this.state.password) {
+      this.setState({error: __('Passwords do not match')});
+    } else {
+      this.setState({error: undefined});
+    }
+  }
+
+  render() {
+    const {className, id, name, verify, error} = this.props.data;
+    const username = document.getElementById('user_login').value;
+
+    return (
+      <div>
+        <CommonForm label={__('Password')} touched={true} error={error}>
+          <ReactPasswordStrength
+            changeCallback={(data) => {this.setState({password: data.password});}}
+            minLength={6}
+            minScore={2}
+            userInputs={[username]}
```

Comment:
@ohadlevy, `user_inputs` array adds strings to [zxcvbn](https://github.com/dropbox/zxcvbn) dictionary.  its meant for user inputs from other fields of the form, so this PR  doesn't break username reference.

Reply:
this only works if the username is entered prior to the initializing the component, you would need to bind to the form via js not via a react mounter props.

Reply:
@ohadlevy  , the username value isn't a prop from react mounter, I'm using the DOM  dynamically when rendering, as you can see in line 25.

Reply:
that only reads it when the component renders, it should be an event of when it changes....